app/commands/thredded/notify_following_users.rb

```diff
@@ -8,29 +10,48 @@ def initialize(post)
     def run
       Thredded.notifiers.each do |notifier|
         notifiable_users = targeted_users(notifier)
-        notifier.new_post(@post, notifiable_users) if notifiable_users.present?
+        notifiable_users = notifiable_users.select do |user|
+          # Create a notification for the user.
+          # If a notification was already created (from another thread/process),
+          # this will return false due to the unique constraint on the table
+          # and the user will be excluded.
+          Thredded::UserPostNotification.create_from_post_and_user(@post, user)
+        end
+        next if notifiable_users.empty?
+        notifier.new_post(@post, notifiable_users)
       end
     end
 
     def targeted_users(notifier)
-      possible_targeted_users.select do |user|
+      users_subscribed_via(notifier).reject do |user|
+        already_notified_user_ids.include?(user.id)
+      end
+    end
+
+    def users_subscribed_via(notifier)
+      subscribed_users.select do |user|
         NotificationsForFollowedTopics
           .detect_or_default(user.thredded_notifications_for_followed_topics, notifier).enabled? &&
           MessageboardNotificationsForFollowedTopics
             .detect_or_default(messageboard_notifier_prefs_by_user_id[user.id], notifier).enabled?
       end
     end
 
-    def possible_targeted_users
-      @possible_targeted_users ||=
-        @post.postable.followers.includes(:thredded_notifications_for_followed_topics).reject { |u| u == @post.user }
+    def subscribed_users
+      @subscribed_users ||= @post.postable.followers.includes(:thredded_notifications_for_followed_topics).reject do |u|
```

Comment:
UncommunicativeVariableName: Thredded::NotifyFollowingUsers#subscribed_users has the variable name 'u'